service/controllers/messages.go

```diff
@@ -221,6 +237,50 @@ func routes(routeObject response, w http.ResponseWriter) {
 				w.Write(jData)
 				TextToSpeech(response.Message + city + " " + state, 0)
 			}
+		} else if strings.ToLower(firstPars) == "meaning" {
+
+			if len(messageArr) == 0 {
+				w.Write([]byte(`{"status": "success", "message": "ENTER: meaning <word>", "result": ""}`))
+			} else {
+
+				word := messageArr[1:]
+				wordStr := strings.Join(word," ")
+
+				result := HandlerMeaning(wordStr)
+				response := processMeaning(result)
+				fmt.Println(len(response))
+
+				if len(word) == 1 && len(response) > 0 {
+
+					responseJSON := jsonResponseMeaning{
+					Status: true,
+					Message: "here is the meaning of the searched word",
+					Result: response,
+					}
+
+					jData, _ := json.Marshal(responseJSON)
+					w.Write(jData)
+					TextToSpeech(responseJSON.Message + " " + wordStr, 0)
```

Comment:
pass ```wordStr``` from the ```filterForSpeech()``` since words with **', ?, !, $** are not supported by the **text-to-speech** API presently. Always pass through the filter before making a word speak!

Reply:
yaa, i will add the function.